front/app/containers/Admin/projects/project/surveyForm/EditButtonWithWarningModal.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react';
+
+import { RouteType } from 'routes';
+
+import usePhase from 'api/phases/usePhase';
+import useFormSubmissionCount from 'api/submission_count/useSubmissionCount';
+import { downloadSurveyResults } from 'api/survey_results/utils';
+
+import useLocale from 'hooks/useLocale';
+
+import EditWarningModal from 'components/admin/SurveyEditWarningModal';
+import Button from 'components/UI/ButtonWithLink';
+
+import { FormattedMessage } from 'utils/cl-intl';
+import clHistory from 'utils/cl-router/history';
+
+import messages from './messages';
+
+interface Props {
+  projectId: string;
+  phaseId: string;
+}
+
+const EditButtonWithWarningModal = ({ projectId, phaseId }: Props) => {
+  const [showEditWarningModal, setShowEditWarningModal] = useState(false);
+  const editFormLink: RouteType = `/admin/projects/${projectId}/phases/${phaseId}/native-survey/edit`;
+  const locale = useLocale();
+  const { data: phase } = usePhase(phaseId);
+  const { data: submissionCount } = useFormSubmissionCount({
+    phaseId,
+  });
+
+  if (!phase || !submissionCount) {
+    return null;
+  }
+
+  const handleDownloadResults = async () => {
+    try {
+      await downloadSurveyResults(locale, phase.data);
+    } catch (error) {
+      // Not handling errors for now
```

Comment:
Was this already doing this in the code that you missed moving over?

Reply:
Yes!